crates/nu-parser/src/lex.rs

```diff
@@ -433,7 +456,32 @@ pub fn lex(input: &str, span_offset: usize) -> (Vec<Token>, Option<ParseError>)
                 Span::new(span_offset + idx, span_offset + idx + 1),
             ));
         } else if *c == '#' {
-            skip_comment(&mut char_indices);
+            let comment_start = *idx + 1;
+            let mut comment = String::new();
+            //Don't copy '#' into comment string
```

Comment:
We may want to skip both the `#` and any leading whitespace, since that white space would just move the text around in the command.

Reply:
That'd let you do:

```
# My echo
def e [arg] {echo $arg}
```

Which I think reads a little better than having to avoid using a space.

Reply:
I agree. Good point. I have updated this PR to include the desired behaviour.

> and any leading whitespace,

That might result in unexpected behavior. Example:
```shell
# The command lets you do: (Space indented)
#    * First thing (Tab indented)
#    * Second thing. (Tab indented)
```
Solution: Not removing the tab here would give the expected behavior.

```shell
#The command lets you do: (No indentation)
# First thing (Space indentation)
# Second thing. (Space indentation)
```
Solution: Giving the user better guidance how to format comments, may solve this problem.

Because of the above mentioned case I have only removed leading spaces.

Reply:
One solution would be to get the number of spaces from the first line, and only remove that many from the other lines, if they're there.

That way your example with the indent could work.

Reply:
Done.